### DISCUSSION TITLE
Screen output on browser laggy

### DISCUSSION THREAD

**Discussion by sw1002** [tesla-android/issue-tracker#157] [General]
Will the screen output via 9.9.0.1 on the tesla browser increase in speed in newer releases or is it an issue of the browser itself?

i am expecing big lags and sometimes it takes 10 seconds after a keypress, is ut possible to up the refresh rate of the web output somehow in settings?

**Comment by adamChrystie** [comment]
Please provide some info about your operating environment. For example, what tesla firmware, type of car you have, MCU version (1,2,3). Intel atom based MCU or AMD ryzen. What version of tesla-android. Do you have a fan cooling the pi cup? Is your pi in a metal case (these are known to hinder Wi-Fi from pi to tesla) closed plastic case , post a picture perhaps.

**Reply by mikegapinski** [reply]
Thanks @electricAntEater32 for dealing with repeating issues on Github, saves me a lot of time! Kudos!

**Comment by adamChrystie** [comment]
I’d say your experience of 10 second lag after a touch event is not normal.

**Comment by brucmod** [comment]
@sw1002  found that placement of the PI4 has a big effect on lag. Try to keep your pi as close to the wifi antennae in your car as possible, Like @electricAntEater32 said, no metal case, plastic only, cooling fan and heat sink. Overclock Pi if possible..  Current systems is pretty darn close to factory response now, but will most likely never be perfect

**Reply by adamChrystie** [reply]
A new version of tesla-android  (2022.45.1) was released last night. Video & touch events  seems faster.  Seems Overclocking is no longer recommended & in install guide the settings were removed. Mike will remove mentions of overclocking soon according a reply to an issue ticket I created today.  Give the new version a try without overclocking in addition to previous tips.

**Reply by sw1002** [reply]
Works smoother for me, disabled over locking , although there still remains a problem with placing the pi cause there is no usb port in the X in the glove box (that was Mike’s suggestion). When I place it in the center console (best place) i get 3 of 4 signal strength on teslas wifi Display, the solution runs slower and laggy, placed on the passenger seat 4/4 and fine.
An external antenna would be very useful for that, but I am not very experienced in soldiering the pi

**Reply by sw1002** [reply]
I found out that it is really a matter of the Pis weak WiFi antenna, even when I place it outside the glovebox on the seat I get 3 of 4 in WiFi. Only when placed next to the right mirror from the inside of the car it steps up to 4/4.
